pkg/pillar/hardware/smart.go

```diff
@@ -80,6 +81,56 @@ func getInfoFromUnknownDisk(diskName, diskType string) *types.DiskSmartInfo {
 	return diskInfo
 }
 
+// getSmartType returns "Pre-fail" if the ATA attribute flag for pre-failure is set,
+// otherwise it returns "Old_age". According to the ATA spec, if bit 0 (prefailure) is set,
+// the attribute is considered a pre-fail attribute.
+// I have found this in smartmontools repo https://github.com/smartmontools/smartmontools.git
+// smartmontools/smartmontools/atacmds.h:164:#define ATTRIBUTE_FLAGS_PREFAILURE(x) (x & 0x01)
+// smartmontools/smartmontools/ataprint.cpp:1302: (ATTRIBUTE_FLAGS_PREFAILURE(attr.flags) ? "Pre-fail" : "Old_age"),
+func getSmartType(flags uint16) string {
+	if flags&0x1 != 0 {
+		return "Pre-fail"
+	}
+	return "Old_age"
+}
+
+func smartAttrMap(id uint8) string {
+	var smartAttrMapping = map[uint8]string{
+		1:   "Raw_Read_Error_Rate",
+		3:   "Spin_Up_Time",
+		4:   "Start_Stop_Count",
+		5:   "Reallocated_Sector_Ct",
+		7:   "Seek_Error_Rate",
+		9:   "Power_On_Hours",
+		10:  "Spin_Retry_Count",
+		12:  "Power_Cycle_Count",
+		177: "Wear_Leveling_Count",
+		179: "Used_Rsvd_Blk_Cnt_Tot",
+		181: "Program_Fail_Cnt_Total",
+		182: "Erase_Fail_Count_Total",
+		183: "Runtime_Bad_Block",
+		187: "Uncorrectable_Error_Cnt",
+		188: "Command_Timeout",
+		190: "Airflow_Temperature_Cel",
+		192: "Power-Off_Retract_Count",
+		193: "Load_Cycle_Count",
+		194: "Temperature_Celsius",
+		195: "ECC_Error_Rate",
+		197: "Current_Pending_Sector",
+		198: "Offline_Uncorrectable",
+		199: "CRC_Error_Count",
+		200: "Multi_Zone_Error_Rate",
+		235: "POR_Recovery_Count",
+		240: "Head_Flying_Hours",
+		241: "Total_LBAs_Written",
+		242: "Total_LBAs_Read",
+	}
+	if name, ok := smartAttrMapping[id]; ok {
+		return name
+	}
+	return "Unknown_Attribute"
```

Comment:
You could return an error here (`func smartAttrMap(id uint8) (string, error)`) so it's easy to the caller to check....

Reply:
There is no error here. The storage device might have a S.M.A.R.T. attribute that we do not know. So, we inform the user that we have a smartAttr.Id that we cannot translate. The Smartctl utility also has similar functionality.